src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -53,4 +55,25 @@ public synchronized void doPost(HttpServletRequest request, HttpServletResponse
 
     response.sendRedirect("/logged_in_homepage.html");
   }
+
+  private User stringToUser(String str) {
```

Comment:
Manually doing serialization/deserialization like this is toilsome and brittle (you'll likely need to add quite a bit more to this as you filter on more fields. Gson (which you touched on briefly in the portfolio project) should be able to handle this use-case more cleanly/scalably.

Let me know if there's some reason I'm missing for why gson wouldn't be a good fit here, happy to discuss!

Reply:
I implemented this, please let me know your thoughts. I still need to add a toString() method so that the strings don't display in Json format on the webpage - I will work on this next.